core/src/main/java/hudson/model/Project.java

```diff
@@ -60,7 +57,7 @@
  * @author Kohsuke Kawaguchi
  */
 public abstract class Project<P extends Project<P,B>,B extends Build<P,B>>
-    extends AbstractProject<P,B> implements SCMTriggerItem, Saveable, ProjectWithMaven, BuildableItemWithBuildWrappers {
```

Comment:
@Vlatombe @jglick  won't this change impact how other plugins? I've done a search on github and many plugins including maven plugin check for this interface.

@Vlatombe did you check the consequences when you prepared this PR?